app/models/user.rb

```diff
@@ -66,7 +68,7 @@ def unset_primary_email
   def set_next_email_as_primary
     return false unless default_email_record
 
-    default_email_record.update_attributes(primary: true)
+    default_email_record.destroyed? || default_email_record.update_attributes(primary: true)
```

Comment:
Yep, I was about to discuss this with you.  Will do this in another PR, thanks !